### DISCUSSION TITLE
osu!lazer merges 2 mania skins.

### DISCUSSION THREAD

**Discussion by mysteGH** [ppy/osu#23355] [Q&A]
Wanted to try out mania on lazer but whenever i try to import skins from stable to lazer it merges 2 skins together, resulting in buggy/weird gameplay elements that i cant really play with. UI scale is set to 1.00x, nothing was changed in skin editor, and skin imported directly from .osk export in stable. below are examples of skin names merging together into 2 skins and what the gameplay looks like. help appreciated! 
![merged skins](https://user-images.githubusercontent.com/96603029/235449171-1990e235-81d4-447e-90f4-3af4612050b0.png)
![osu_2023-05-01_21-55-51](https://user-images.githubusercontent.com/96603029/235449213-21282237-010e-4a38-9995-e9048fb7787d.jpg)

And this is what it looks like in stable:
![screenshot060](https://user-images.githubusercontent.com/96603029/235449807-0433a38c-f424-4d97-b1f6-7d304bab0839.png)

**Comment by nagi-desuuu** [comment]
Have you tried removing the problematic skin first and then reimporting the clean one?

**Reply by mysteGH** [reply]
the issue was with *all* skins, sorry i forgot to mention that. I also tried exporting a clean .osk skin from stable, and importing that into lazer and still with no success. However, with the same osk, i tried it on a different pc and the import worked perfectly fine. It's gotta be something going on with my pc.

**Comment by peppy** [comment]
There's no way that lazer can "merge" skins together, so you're definitely seeing something else going on here.

That said, mania skinning is very complex and I wouldn't be surprised if exporting from lazer has issues on stable's end in some weird edge case way that only affects your skin (potentially directory structure not being kept, or case sensitivity, or something else).

You're going to need to attach the skins in their good state (ie. before you imported to lazer) for us to be able to help you.

**Reply by mysteGH** [reply]
it's not only mania skins though, it's every skin that was imported from stable to lazer that are "merging" together (i dont know how else to describe it, it's probably something else, i just assumed they're merged because on lazer, skins show with 2 names under 1 skin and there are elements from both skins in one) im pretty bad at describing things so sorry if it made no sense. 

I'll attach my main mania skin in .osk form below. the issue on my end is that the .osk is exported from osu!stable (all skinning elements are fine, and the skin is only titled "keii_bar" on stable for me), but when i try to import this .osk into lazer it renames to "keii_bar _ keiichi__" (keiichi and keii_bar are two different skins)

I may be doing something wrong in the importing process so if need be i can also attach a video later on how i've been trying to import to help figure it out.

also thanks for reply =D.
[stable export skin.zip](https://github.com/ppy/osu/files/11380242/stable.export.skin.zip)

**Reply by Theighlin** [reply]
To avoid conflicts, skin names in lazer are folder name + skin.ini name (or viceversa)

**Reply by peppy** [reply]
To me, this just sounds like the user has a lot of skins on stable which have different naming to lazer, and is getting confused by this.
